sydent/config/__init__.py

```diff
@@ -73,3 +205,97 @@ def parse_from_config_parser(self, cfg: ConfigParser) -> bool:
         #       user has asked for this specifially (e.g. on first
         #       run only, or when specify --generate-config)
         return self.crypto.save_key
+
+    def parse_config_file(self, config_file: str) -> None:
+        """
+        Parse the given config from a filepath, populating missing items and
+        sections
+
+        :param config_file: the file to be parsed
+        """
+        # If the config file doesn't exist, prepopulate the config object
+        # with the defaults, in the right section.
+        #
+        # Otherwise, we have to put the defaults in the DEFAULT section,
+        # to ensure that they don't override anyone's settings which are
+        # in their config file in the default section (which is likely,
+        # because sydent used to be braindead).
+        use_defaults = not os.path.exists(config_file)
+
+        cfg = ConfigParser()
+        for sect, entries in CONFIG_DEFAULTS.items():
+            cfg.add_section(sect)
+            for k, v in entries.items():
+                cfg.set(DEFAULTSECT if use_defaults else sect, k, v)
+
+        cfg.read(config_file)
+
+        # Logging is configured in cfg, but these options must be parsed first
+        # so that we can log while parsing the rest
+        setup_logging(cfg)
```

Comment:
it's a bit weird/unexpected that `parse_config_file` has this side-effect. I kinda see how we got here, and let's not change it for now, but I'd like us to think about changing it once this PR lands.

Reply:
I've added a note to the docstring for it, but I agree that there is probably a better way